From df5cb96d029f9e46a58e6f7f6fb175abe459730e Mon Sep 17 00:00:00 2001
From: Pete Beardmore <pete.beardmore@msn.com>
Date: Wed, 20 Jun 2018 16:32:14 +0100
Subject: [fix] hackerrank, 'time delta', python solution

-feb idx off-by-one error in leap year application
-duplicated minute output as opposed to seconds in to_string
---
 .../time.delta.py                               | 17 ++++++++++-------
 1 file changed, 10 insertions(+), 7 deletions(-)

diff --git a/hackerrank/python.-.date.and.time.-.medium.-.time.delta/time.delta.py b/hackerrank/python.-.date.and.time.-.medium.-.time.delta/time.delta.py
index 29202eb..3a15fab 100644
--- a/hackerrank/python.-.date.and.time.-.medium.-.time.delta/time.delta.py
+++ b/hackerrank/python.-.date.and.time.-.medium.-.time.delta/time.delta.py
@@ -30,7 +30,7 @@ def day_of_year(day, mon, year):
     days = day
     if mon > 0:
         days += sum(list(month_days.values())[0:(mon - 1) + 1]);
-    if is_leap(year) and mon > 2:
+    if is_leap(year) and mon > 1:
         days += 1
     return days
 
@@ -135,7 +135,7 @@ class zdate:
         return res
 
     def to_string(self):
-        return ("%d %s %d %02d:%02d:%02d" % (self.d, self.mn, self.y, self.th, self.tm, self.tm))
+        return ("%d %s %d %02d:%02d:%02d" % (self.d, self.mn, self.y, self.th, self.tm, self.ts))
 
 # Complete the time_delta function below.
 def time_delta(dt1, dt2):
@@ -145,12 +145,12 @@ def time_delta(dt1, dt2):
     zd1 = zdate(dt1)
     zd2 = zdate(dt2)
 
-    print("[zd1] %s" % (zd1.to_string()))
+    print("[zd1] %s, dy: %d" % (zd1.to_string(), zd1.dy))
     zd1.utc_adjust()
-    print("[zd1] %s [utc]" % (zd1.to_string()))
-    print("[zd2] %s" % (zd2.to_string()))
+    print("[zd1] %s, dy: %d [utc]" % (zd1.to_string(), zd1.dy))
+    print("[zd2] %s, dy: %d" % (zd2.to_string(), zd2.dy))
     zd2.utc_adjust()
-    print("[zd2] %s [utc]" % (zd2.to_string()))
+    print("[zd2] %s, dy: %d [utc]" % (zd2.to_string(), zd2.dy))
 
     if zd1.is_after(zd2):
         zd1, zd2 = zd2, zd1
@@ -173,7 +173,10 @@ def time_delta(dt1, dt2):
     print("[zd2] d: %s, m: %d, dy: %d" % (zd2.to_string(), zd2.m, zd2.dy))
 
     print("[result] %d" % (s_delta))
-    print("[target] 413962293")
+    if str(s_delta)[0] == "4":
+        print("[target] 413962293")
+    else:
+        print("[target] 1155677269")
 
     return str(s_delta)
 
-- 
2.23.0

